guides/isolation.md

```diff
@@ -0,0 +1,178 @@
+
+```elixir
+def start(_type, _args) do
+  children = [
+    MyApp.Repo,
+    MyApp.Endpoint,
+    {Oban, name: ObanA, repo: MyApp.Repo},
+    {Oban, name: ObanB, repo: MyApp.Repo, prefix: "special"},
+    {Oban, name: ObanC, repo: MyApp.Repo, prefix: "private"}
+  ]
+
+  Supervisor.start_link(children, strategy: :one_for_one, name: MyApp.Supervisor)
+end
+```
+
+When you do this, you'll have to use the correct Oban supervisor name when performing Oban-related
+operations. You'll see that most functions in the `Oban` module, for example, take an optional
+first argument which represents the name of the Oban supervisor. By default, that's `Oban`, which
+is why this works if you don't explicitly start an Oban supervisor in your application:
+
+```elixir
+Oban.insert(MyApp.Worker.new(%{}))
+```
+
+In the example above, with `ObanA`/`ObanB`/`ObanC`, you can specify which Oban instance you want
+to use for scheduling by passing its name in:
+
+```elixir
+Oban.insert(ObanB, MyApp.Worker.new(%{}))
+```
+
+### Umbrella Apps
+
+If you need to run Oban from an umbrella application where more than one of the child apps need to
+interact with Oban, you may need to set the `:name` for each child application that configures
+Oban.
+
+For example, your umbrella contains two apps: `MyAppA` and `MyAppB`. `MyAppA` is responsible for
+inserting jobs, while only `MyAppB` actually runs any queues.
+
+Configure Oban with a custom name for `MyAppA`:
+
+```elixir
+config :my_app_a, Oban,
+  name: MyAppA.Oban,
+  repo: MyApp.Repo
+```
+
+Then configure Oban for `MyAppB` with a different name and different options:
+
+```elixir
+config :my_app_b, Oban,
+  name: MyAppB.Oban,
+  repo: MyApp.Repo,
+  queues: [default: 10]
+```
+
+Now, use the configured name when calling functions like `Oban.insert/2`, `Oban.insert_all/2`,
+`Oban.drain_queue/2`, and so on, to reference the correct Oban process for the current
+application.
+
+```elixir
+Oban.insert(MyAppA.Oban, MyWorker.new(%{}))
+Oban.insert_all(MyAppB.Oban, multi, :multiname, [MyWorker.new(%{})])
+Oban.drain_queue(MyAppB.Oban, queue: :default)
+```
+
+## Database Isolation
+
+Let's look at a few options for isolating or scoping Oban database queries.
+
+### Database Prefixes
+
+Oban supports namespacing through **PostgreSQL schemas**, also called "prefixes" in Ecto. With
+prefixes, your job table can reside outside of your primary schema (usually `public`) and you can
+have multiple separate job tables.
+
+To use a prefix you first have to specify it within your migration:
+
+```elixir
+defmodule MyApp.Repo.Migrations.AddPrefixedObanJobsTable do
+  use Ecto.Migration
+
+  def up do
+    Oban.Migrations.up(prefix: "private")
+  end
+
+  def down do
+    Oban.Migrations.down(prefix: "private")
+  end
+end
+```
+
+The migration will create the `private` schema and all Oban-related tables, functions, and
+triggers within that schema. With the database migrated, you'll then specify the prefix in your
+configuration:
```

Comment:
There aren't any triggers or additional functions created with the current migrations:

```suggestion
The migration will create the `private` schema and all Oban-related tables within that schema.
With the database migrated, you'll then specify the prefix in your configuration:
```